### PR TITLE
feat(ui): add social dropdown list for desktop

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add social dropdown list for desktop
May-15-2022 20-41-44

Also clean up all the social component styles. Break theme into different CSS module files.

## Linear Issue Reference

log-2431

## Testing

test lodally
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.module.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+.socialDropDown {
+  position: absolute;
+  min-width: 208px;
+  transform: translateY(-100%) scale(0);
+  transform-origin: 12px bottom;
+  opacity: 0%;
+  transition: transform 0.1s, opacity 0.1s;
+}
+
+.social_logo {
```
Comment (demonzoo) [inline packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.module.scss:12]: nit: `socialLogo`?

File: packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      });
+    }
+  }, [anchorRef]);
+
+  return (
+    <Dropdown
+      role="dropdown"
+      isOpen={isOpen}
+      className={styles.socialDropDown}
+      style={{ content: contentStyle }}
```
Comment (demonzoo) [inline packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx:64]: Just curious: Isn't `content only` available in `:before` or `:after` pseudo classes?

Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx:64]: This attribute is ReactModal.style, which accepts two params content and overlay.  not HTML style

Comment (demonzoo) [reply packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx:64]: ok

File: packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      setContentStyle({
+        left,
+        top: top - 8,
+      });
+    }
+  }, [anchorRef]);
+
+  return (
+    <Dropdown
+      role="dropdown"
```
Comment (demonzoo) [inline packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx:61]: nit: Would it be better to put this role in Dropdown component itself? Also, do you think it necessary to put this in the `console` dropdown component as well?

**Review by demonzoo — APPROVED** [review]

File: packages/ui/src/components/BrandingHeader/index.module.scss [hunk 4]
```diff
@@ -4,23 +4,45 @@ $logo-height: 60px;
...
 }
 
 .logo {
   height: $logo-height;
   width: auto;
   @include _.image-align-center;
+}
+
+
+:global(body.mobile) {
```
Comment (wangsijie) [inline packages/ui/src/components/BrandingHeader/index.module.scss:17]: need to rebase?

Comment (simeng-li) [reply packages/ui/src/components/BrandingHeader/index.module.scss:17]: yes

File: packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx (hunk 3 above)
Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SocialSignInDropdown.tsx:61]: Actually, it was a legacy code. It was intended to limit the ReactMode animations style only apply to this component as I have to use the global ReactMode provided classnames. No longer needed now. I'll delete it.

**Pull request merged**